src/wxWidgets.cpp

```diff
@@ -537,45 +537,26 @@ void CText::Draw( CDxf* dxf )
         theDC->DrawText( draw.text, draw.x1, draw.y1 );
     }
 }
+
 void CSpline::Draw( CDxf* dxf )
 {
-    wxPoint spline_points[MAXPOINTS];
     cDrawPrimitiveData draw( dxf );
 
     wxColour c(draw.color);
     theDC->SetBrush(wxBrush(c));
     theDC->SetPen(wxPen(c, draw.thick));
 
+    std::vector<wxPoint> points;
+
     // loop over drawing primitives
     while( getDraw( draw ) )
     {
-        if( dxf->wxwidgets() && m_ControlPointCount>0  )
-        {
-            // use wxwidgets spline method for control point spline TID6
-
-            if( ! draw.index_curve )
-            {
-                // collect control point
-                wxPoint p( draw.x1, draw.y1 );
-                spline_points[ draw.index-1] = p;
-            }
-            else
-            {
-                // all contol points
-                // use wxwidget spline method
-                theDC->DrawSpline(
-                    m_ControlPointCount,
-                    spline_points );
-            }
-        }
-        else
-        {
-            // use wxwidgets drawing primitive for spline
-
-            theDC->DrawLine( draw.x1, draw.y1, draw.x2, draw.y2 );
-        }
+        // use wxwidgets drawing primitive for spline
+        points.emplace_back(wxPoint(draw.x1, draw.y1));
     }
+    theDC->DrawSpline(points.size(), points.data());
```

Comment:
This is wrong.  Have you looked at what it does? https://docs.wxwidgets.org/3.2/classwx_d_c.html#ad72d38c75ac6b3f80c1f957dd94089fa

The interpolating points of the regenerated spline have already been calculated and stored in draw.x1, draw.y1.  Now you are invoking another spline calculator to generate  a new spline using the results of the previous calculations.

This will double the time taken to draw the spline, perhaps even more since you are passing in so many "control points".  Did you do any timing tests before committing this?

You should be using DrawLines() ( https://docs.wxwidgets.org/3.2/classwx_d_c.html#abae438499fdfdc102e7be7629e8f8c46 ) which will run much faster and produce better results.